edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/SubscriptionProcessor.cs

```diff
@@ -142,11 +144,29 @@ async Task ProcessSubscriptionByIdentity(IIdentity identity)
             }
         }
 
-        async void ClientCloudConnectionEstablished(object sender, IIdentity identity)
+        void ClientConnectionToCloudEstablished(object sender, IIdentity identity)
         {
+            this.ClientConnectionEstablished(identity, true);
+        }
+
+        void ClientConnectionToEdgeHubEstablished(object sender, IIdentity identity)
+        {
+            this.ClientConnectionEstablished(identity, false);
+        }
+
+        async void ClientConnectionEstablished(IIdentity identity, bool connectedToCloud)
+        {
+            if (connectedToCloud)
```

Comment:
nit: connectedToCloud could be passed to the Events method so the if can be inside it